backend/src/Service/StoreOwnerBranch/StoreOwnerBranchService.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace App\Service\StoreOwnerBranch;
+
+use App\AutoMapping;
+use App\Constant\StoreOwnerBranch\StoreOwnerBranch;
+use App\Entity\StoreOwnerBranchEntity;
+use App\Manager\StoreOwnerBranch\StoreOwnerBranchManager;
+use App\Request\StoreOwnerBranch\StoreOwnerBranchCreateRequest;
+use App\Request\StoreOwnerBranch\StoreOwnerBranchDeleteRequest;
+use App\Response\StoreOwnerBranch\StoreOwnerBranchResponse;
+
+class StoreOwnerBranchService
+{
+    private $autoMapping;
+    private $storeOwnerBranchManager;
+
+    public function __construct(AutoMapping $autoMapping, StoreOwnerBranchManager $storeOwnerBranchManager)
+    {
+        $this->autoMapping = $autoMapping;
+        $this->storeOwnerBranchManager = $storeOwnerBranchManager;
+    }
+
+    /**
+     * @param StoreOwnerBranchCreateRequest $request
+     * @return StoreOwnerBranchResponse|null
+     */
+    public function createBranch(StoreOwnerBranchCreateRequest $request): ?StoreOwnerBranchResponse
+    {
+        $branch = $this->storeOwnerBranchManager->createBranch($request);
+
+        return $this->autoMapping->map(StoreOwnerBranchEntity::class, StoreOwnerBranchResponse::class, $branch);
+    }
+
+    /**
+     * @param $request
+     * @return StoreOwnerBranchResponse|string
+     */
+    public function updateBranch($request): string|StoreOwnerBranchResponse
+    {
+        $result = $this->storeOwnerBranchManager->updateBranch($request);
+        if($result == StoreOwnerBranch::BRANCH_NOT_FOUND){
+
+            return StoreOwnerBranch::BRANCH_NOT_FOUND;
+        }
+
+        return $this->autoMapping->map(StoreOwnerBranchEntity::class, StoreOwnerBranchResponse::class, $result);
+    }
+
+    /**
+     * @param StoreOwnerBranchDeleteRequest $request
+     * @return StoreOwnerBranchResponse|string
+     */
+    public function deletebranch(StoreOwnerBranchDeleteRequest $request): string|StoreOwnerBranchResponse
+    {
+        $result = $this->storeOwnerBranchManager->deletebranch($request);
+        if($result == StoreOwnerBranch::BRANCH_NOT_FOUND){
+
+            return StoreOwnerBranch::BRANCH_NOT_FOUND;
+        }
+
+        return $this->autoMapping->map(StoreOwnerBranchEntity::class, StoreOwnerBranchResponse::class, $result);
+    }
+
+    /**
+     * @param $storeOwnerId
+     * @return array
+     */
+    public function getAllBranches($storeOwnerId):array
+    {
+        $response = [];
+
+        $items = $this->storeOwnerBranchManager->getAllBranches($storeOwnerId);
```

Comment:
We may use more clarified name for variables instead of _item_.

Reply:
I agree.